docs/docs/features/testing-mcp-server.md

```diff
@@ -0,0 +1,337 @@
+- **Validate Tool Responses**: Confirm that tools return expected data
+- **Test RBAC Permissions**: Verify Kubernetes API access works properly
+
+Once your MCP server passes these tests, you can confidently connect it to Headlamp-KAITO via the [MCP Tool Calling](./mcp-tool-calling.md) interface.
+
+## MCP Kubernetes Integration Guide (with RBAC)
+
+This guide walks you through deploying the MCP Inspector UI and Kubernetes MCP Server inside a Kubernetes cluster with role-based access control (RBAC)-based access to the Kubernetes API.
+
+## Prerequisites
+
+- A working Kubernetes cluster (e.g., AKS, EKS, GKE, or Minikube)
+- kubectl configured for the cluster
+
+## 1. Create a Service Account and RBAC for the Kubernetes MCP Server
+
+The kubernetes-mcp service account will be used only by the Kubernetes MCP Server, allowing it to read basic cluster resources via the Kubernetes API.
+
+[Download kubernetes-mcp-0.1.0.tgz](../../static/kubernetes-mcp-0.1.0.tgz) and install with:
+
+```bash
+helm install kubernetes-mcp ./kubernetes-mcp-0.1.0.tgz
+```
+
+After completing this step, proceed directly to **[Step&nbsp;2: Deploy Kubernetes MCP Server](#2-deploy-kubernetes-mcp-server)** on this page. If you experience any issues, manually apply the RBAC configuration files by following steps a, b, and c below.
+
+### a. mcp-serviceaccount.yaml
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: kubernetes-mcp
+  namespace: default
+```
+
+### b. mcp-clusterrole.yaml
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: kubernetes-mcp-role
+rules:
+  - apiGroups: ['']
+    resources: ['pods', 'services', 'nodes']
+    verbs: ['get', 'list', 'watch']
+  - apiGroups: ['apps']
+    resources: ['deployments']
+    verbs: ['get', 'list', 'watch']
+```
+
+### c. mcp-clusterrolebinding.yaml
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: kubernetes-mcp-binding
+subjects:
+  - kind: ServiceAccount
+    name: kubernetes-mcp
+    namespace: default
+roleRef:
+  kind: ClusterRole
+  name: kubernetes-mcp-role
+  apiGroup: rbac.authorization.k8s.io
+```
+
+Apply all:
+
+```bash
+kubectl apply -f mcp-serviceaccount.yaml
+kubectl apply -f mcp-clusterrole.yaml
+kubectl apply -f mcp-clusterrolebinding.yaml
+```
+
+## 2. Deploy Kubernetes MCP Server
+
+The Kubernetes MCP Server handles AI tool requests and must be deployed using the `mcp-service` account created above.
+
+### kubernetes-mcp-server.yaml
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kubernetes-mcp-server
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: kubernetes-mcp-server
+  template:
+    metadata:
+      labels:
+        app: kubernetes-mcp-server
+    spec:
+      serviceAccountName: kubernetes-mcp
+      automountServiceAccountToken: true
+      containers:
+        - name: mcp
+          image: ghcr.io/chloe608/kubernetes-mcp-server
```

Comment:
Could you push the container image to `ghcr.io/kaito-project/headlamp-kaito` namespace instead?